doc/specs/vulkan/chapters/memory.txt

```diff
@@ -1721,6 +1721,7 @@ ifdef::VK_KHX_device_group[]
   * [[VUID-vkMapMemory-memory-00683]]
     pname:memory must: not have been allocated with multiple instances.
 endif::VK_KHX_device_group[]
+  * pname:ppData must: be a valid pointer to a pointer
```

Comment:
Isn't this what would be generated automatically by your script change anyway?

Reply:
TBH `void*` with `len` should have perhaps been `uint8_t`, but I guess too late to change that (till Vulkan 2.0).

Reply:
Eh. I mean we at least differentiated in the parameter names (size vs. count) - there are times when you just don't know what the data is in C, and I think for the most part we've limited our use of void* to just those. Don't necessarily think that needs changing.

> Perhaps the vk.xml parsing script should throw error/exception when there is a void* without len && without noautovalidity.

Also I think perhaps a void* without a length is probably something we should catch. Could you raise an issue about it? (or a PR if you're up to it :P)

Reply:
Well, the `len` on `void*` here implies bytes. `void` is not byte, neither really is `char` semantically (of course that's how people still coded it for decades anyway). It does not probably even have good alignment to be cast to anything else (so I know what data it is &mdash; bytes &mdash; unless otherwisely said by the spec).

> Could you raise an issue about it? (or a PR if you're up to it :P)

Sure. I could do it as part of this PR, or do you think it is getting bit overloaded?
Oh, and plese tell me if you would like something here unsquashed or perhaps as a separate PR.

Reply:
I'd suggest a separate PR, otherwise I think this is going to get a bit difficult to review (plus higher chance of conflicts with later revisions if this drags, etc.).

Reply:
FYI: Sugested as PR #562.